### DISCUSSION TITLE
change file name in RotatingFileSink

### DISCUSSION THREAD

**Discussion by lioriz1** [SpriteOvO/spdlog-rs#49] [Q&A] [answered]
I'm using RotatingFileSink for rotating log files.
Sink creation:

    let path_by_size: PathBuf = PathBuf::from("/log/records/my_telemetry.log");
    
    let by_size: Arc<RotatingFileSink> = Arc::new(
        RotatingFileSink::builder()
            .base_path(path_by_size)
            .max_files(64)
            .rotation_policy(RotationPolicy::FileSize(constants::LOG_FILE_MAX_SIZE))
            .rotate_on_open(true)
            .build()?,
    );
    
    let logger: Logger = Logger::builder().sink(by_size).build()?;

I'm getting `my_telemetry.log`, `my_telemetry_1.log`, `my_telemetry_2.log` ... `my_telemetry_63.log`, rotating as expected.

But, I need the files to have unique names and I didn't find a way to do it in the `spdlog-rs`.

In our `CPP` project, the `spdlog::sinks::rotating_file_sink` can have unique file names using `file_event_handlers.after_close`:

    static std::shared_ptr<spdlog::sinks::rotating_file_sink_mt> CreateRotatingFileSink(const std::string &dir_name) {
        spdlog::file_event_handlers file_event_handlers;
        file_event_handlers.after_close = [dir_name](const spdlog::filename_t &filename) {
            static thread_local boost::circular_buffer<std::string> log_files_names(GetMaxLogsFile());
            static thread_local int i = 1;

            bool exist_max_logs_file = log_files_names.size() == log_files_names.capacity();
            if (exist_max_logs_file) {
                std::string oldest_log_file = log_files_names.front();
                std::remove(oldest_log_file.c_str());
            }

            auto now = SystemTime::Now();
            std::string new_filename = std::string(dir_name + std::to_string(i++) + "." +
                                              std::to_string(now.time_since_epoch().count()) + LOG_FILE_EXTENSION);
            log_files_names.push_back(new_filename);

            std::rename(filename.c_str(), new_filename.c_str());
        };

        return std::make_shared<spdlog::sinks::rotating_file_sink_mt>(dir_name + "current_file" + LOG_FILE_EXTENSION,
                                                                      GetMaxSizeFile(),
                                                                      1,
                                                                      false,
                                                                      file_event_handlers);
    }

Is there a similar way to make unique file names in the RotatingFileSink?

**Comment by SpriteOvO** [comment, marked as answer]
Currently it's impossible. I'm considering adding a method `custom_rename` on `RotatingFileSinkBuilder`.

UPDATE: Tracking issue #53.

**Reply by lioriz1** [reply]
Ok
Will need to think about a workaround for this issue for now.
Waiting for this release.
Thanks

**Reply by Lancern** [reply]
@lioriz1 If you are interested in this feature, feel free to open a new feature request in the issues so we can make further discussions and track the development progress there. Thanks.

**Comment by SpriteOvO** [comment]
We plan to implement some event hooks (e.g. `after_close`) for `FileSink` and `RotatingFileSink`, as discussion in #51 points out their necessity.

**Comment by SpriteOvO** [comment]
Opened a issue #53 to track the progress.
